From 8c77b60f34c176430f49380e3b450165d5a12324 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Wed, 20 Feb 2019 18:01:19 -0600
Subject: [PATCH] crypto/o_time.c: Workaround gmtime_r() bug on VxWorks

gmtime_r() always returns 0 on VxWorks, so ignore it's return value on
this platform.

OpenSSL_1_0_2k and earlier ignored gmtime_r() retval on all platforms.
The code changed in OpenSSL_1_0_2l (commit b70dc3a66d1) and broke
VxWorks builds. This change reverts to the old behavior only on VxWorks.

Upstream-Status: Inappropriate [VxWorks support]
---
 crypto/o_time.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/crypto/o_time.c b/crypto/o_time.c
index 61927439ea..eaeadb25f1 100755
--- a/crypto/o_time.c
+++ b/crypto/o_time.c
@@ -106,8 +106,13 @@ struct tm *OPENSSL_gmtime(const time_t *timer, struct tm *result)
     struct tm *ts = NULL;
 
 #if defined(OPENSSL_THREADS) && !defined(OPENSSL_SYS_WIN32) && !defined(OPENSSL_SYS_OS2) && (!defined(OPENSSL_SYS_VMS) || defined(gmtime_r)) && !defined(OPENSSL_SYS_SUNOS)
+#if defined(OPENSSL_SYS_VXWORKS)
+    /* XXX gmtime_r() always returns NULL on VxWorks, ignore it */
+    gmtime_r(timer, result);
+#else
     if (gmtime_r(timer, result) == NULL)
         return NULL;
+#endif
     ts = result;
 #elif !defined(OPENSSL_SYS_VMS) || defined(VMS_GMTIME_OK)
     ts = gmtime(timer);
-- 
2.20.1

